src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE = 5;
+    private static final int MAX_VALUE = 10;
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURE);
+
+        switch (index) {
+            case 0:
+                Figure circle = new Circle(supplier.getRandomColor(), random.nextInt(MAX_VALUE));
+                circle.getArea();
```

Comment:
Remove the `circle.getArea()` and `circle.draw()` method calls here, as they should not be called within this function. The same applies to lines 25, 33, 38, and 45.